graphql_compiler/deserialization.py

```diff
@@ -0,0 +1,253 @@
+
+    Returns:
+        a value of the type produced by the parser of the expected type:
+            GraphQLDate: datetime.date
+            GraphQLDateTime: datetime.datetime with tzinfo=pytz.utc
+            GraphQLFloat: float
+            GraphQLDecimal: decimal.Decimal
+            GraphQLInt: int
+            GraphQLString: str
+            GraphQLBoolean: bool
+            GraphQLID: str
+
+    Raises:
+        GraphQLInvalidArgumentError: if the argument value was not of the expected type.
+    """
+    # Explicitly disallow passing boolean values for non-boolean types.
+    if isinstance(value, bool) and not is_same_type(GraphQLBoolean, expected_type):
+        raise GraphQLInvalidArgumentError("")
+    try:
+        return convert_scalar_value_to_graphql_type(expected_type, value)
+    except (ValueError, TypeError) as e:
+        raise GraphQLInvalidArgumentError("Error parsing argument {}: {}".format(name, e))
+
+
+def deserialize_argument(name: str, expected_type: QueryArgumentGraphQLType, value: Any,) -> Any:
+    """Deserialize a serialized GraphQL argument.
+
+    Passing arguments via jsonrpc, or via the GUI of standard GraphQL editors is tricky because
+    json does not support certain types like Date, Datetime, Decimal, and also confuses floats
+    for integers if there are no decimals. This function takes in a value and converts it to a
+    standard python representation.
+
+    Below are examples of accepted encodings of all the types:
+        GraphQLDate: "2018-02-01"
+        GraphQLDateTime: "2018-02-01T05:11:54Z"
+        GraphQLFloat: 4.3, "5.0", 5
+        GraphQLDecimal: "5.00000000000000000000000000001"
+        GraphQLInt: 4, "3803330000000000000000000000000000000000000000000"
+        GraphQLString: "Hello"
+        GraphQLBoolean: True, 1, "1", "True", "true"
+        GraphQLID: "13d72846-1777-6c3a-5743-5d9ced3032ed"
+        GraphQLList(GraphQLInt): [1, 2, 3]
+
+    Args:
+        name: he name of the argument. It will be used to provide a more descriptive error
+              message if an error is raised.
+        expected_type: the GraphQL type. All GraphQLNonNull type wrappers are stripped.
+        value: object that can be interpreted as being of that type.
+
+    Returns:
+        a value of the type produced by the parser of the expected type:
+            GraphQLDate: datetime.date
+            GraphQLDateTime: datetime.datetime with tzinfo=pytz.utc
+            GraphQLFloat: float
+            GraphQLDecimal: decimal.Decimal
+            GraphQLInt: int
+            GraphQLString: str
+            GraphQLBoolean: bool
+            GraphQLID: str
+            GraphQLList: list of the inner type
+
+    Raises:
+        GraphQLInvalidArgumentError: if the argument value was not of the expected type.
+    """
+    stripped_type = strip_non_null_from_type(expected_type)
+    if isinstance(stripped_type, GraphQLList):
+        if not isinstance(value, list):
```

Comment:
What do you think of moving this logic "downward" and creating a more generic `convert_value_to_graphql_type` function and expose that function as public API instead of `convert_scalar_value_to_graphql_type`? (If you follow my suggestion above this would be `deserialize_value`).